### PR TITLE
support for certificate creation on Nokia SR Linux nodes 

### PR DESCRIPTION
**Pull request by hellt**
This PR adds support for Nokia SR Linux to be provisioned with self-signed certificates defined in a topology file.

The enclosed topology file (examples/2node-srl-with-cert.pb.txt) contains an example of the topo.

### CODE REVIEW

**Comment by hellt** [comment]
@marcushines this is ready to be reviewed.
to test this out on a kind cluster the following procedure should be followed https://github.com/srl-labs/kne-controller#testing-with-kind

File: topo/node/srl/srl.go [hunk 1]
```diff
@@ -28,15 +37,52 @@ func New(pb *topopb.Node) (node.Implementation, error) {
...
+	}
+	for e := range w.ResultChan() {
+		p := e.Object.(*corev1.Pod)
+		if p.Status.Phase == corev1.PodRunning {
+			break
+		}
+	}
+	log.Infof("%s - pod running.", n.pb.Name)
+
+	err = n.SpawnCLIConn(ni.Namespace())
```
Comment (marcushines) [inline topo/node/srl/srl.go:72]: nit: 
```
if err := n.SpawnCLIConn(ni.Namespace()); err != nil {
  return err
}
```

File: topo/node/srl/srl.go [hunk 2]
```diff
@@ -28,15 +37,52 @@ func New(pb *topopb.Node) (node.Implementation, error) {
...
+	if err != nil {
+		return err
+	}
+
+	defer n.cliConn.Close()
+
+	err = srlinux.AddSelfSignedServerTLSProfile(n.cliConn, selfSigned.CertName, false)
```
Comment (marcushines) [inline topo/node/srl/srl.go:79]: same as above

File: topo/node/srl/srl.go [hunk 3]
```diff
@@ -164,6 +210,69 @@ func defaults(pb *topopb.Node) *topopb.Node {
...
+	if err != nil {
+		return err
+	}
+
+	// jacked up PtyWidth to allow for long strings such as cert and key to not break the terminal
+	d.Transport.BaseTransportArgs.PtyWidth = 5000
+
+	n.cliConn = d
+
+	err = n.PatchCLIConnOpen(ns)
```
Comment (marcushines) [inline topo/node/srl/srl.go:230]: same

File: topo/node/srl/srl.go [hunk 4]
```diff
@@ -164,6 +210,69 @@ func defaults(pb *topopb.Node) *topopb.Node {
...
+	if err != nil {
+		n.cliConn = nil
+
+		return err
+	}
+
+	err = n.WaitCLIReady()
```
Comment (marcushines) [inline topo/node/srl/srl.go:237]: same

Comment (hellt) [reply topo/node/srl/srl.go]: addressed in c57d0ce

File: topo/node/srl/srl.go (hunk 3 above)
Comment (hellt) [reply topo/node/srl/srl.go]: addressed in c57d0ce

**Review by marcushines — APPROVED** [review]

**Pull request merged**